include/picongpu/particles/collision/binary/RelativisticBinaryCollision.hpp

```diff
@@ -0,0 +1,513 @@
+                            // feenableexcept(FE_INVALID | FE_OVERFLOW | FE_DIVBYZERO);
+                            float_COLL const normalizedWeight0
+                                = precisionCast<float_COLL>(par0[weighting_]) / WEIGHT_NORM_COLL;
+                            float_COLL const normalizedWeight1
+                                = precisionCast<float_COLL>(par1[weighting_]) / WEIGHT_NORM_COLL;
+
+                            float3_COLL const labMomentum0
+                                = precisionCast<float_COLL>(par0[momentum_]) / normalizedWeight0;
+                            float3_COLL const labMomentum1
+                                = precisionCast<float_COLL>(par1[momentum_]) / normalizedWeight1;
+                            float_COLL const mass0 = precisionCast<float_COLL>(picongpu::traits::attribute::getMass(
+                                particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                par0));
+                            float_COLL const mass1 = precisionCast<float_COLL>(picongpu::traits::attribute::getMass(
+                                particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                par1));
+                            float_COLL const charge0
+                                = precisionCast<float_COLL>(picongpu::traits::attribute::getCharge(
+                                    particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                    par0));
+                            float_COLL const charge1
+                                = precisionCast<float_COLL>(picongpu::traits::attribute::getCharge(
+                                    particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE,
+                                    par1));
+                            float_COLL const gamma0 = picongpu::gamma<float_COLL>(labMomentum0, mass0);
+                            float_COLL const gamma1 = picongpu::gamma<float_COLL>(labMomentum1, mass1);
+
+                            //                            PMACC_DEVICE_ASSERT_MSG(
+                            //                                math::sqrt(pmacc::math::abs2(labMomentum0)) / gamma0 /
+                            //                                mass0 < c, "labVelocity0: %e",
+                            //                                sqrt(pmacc::math::abs2(labMomentum0)) / gamma0 / mass0);
+                            //                            PMACC_DEVICE_ASSERT_MSG(
+                            //                                math::sqrt(pmacc::math::abs2(labMomentum1)) / gamma1 /
+                            //                                mass1 < c, "labVelocity1: %e",
+                            //                                sqrt(pmacc::math::abs2(labMomentum1)) / gamma1 / mass1);
+                            // [Perez 2012] (1)
+                            // todo:: maybe this also with doubles?
```

Comment:
what do you mean with doubles? Again a safeguard against `float_COLL` being `float_32`?